lib/show.gi

```diff
@@ -0,0 +1,70 @@
+#############################################################################
+##
+#W  show.gi
+##
+##  Contains implementations of standard functions for outputting objects
+##  to streams.
+##
+##  These functions should replace all the View/Print/Display variants, and
+##  in particular get rid of a lot of code duplication, when there are separate
+##  implementations for View and ViewString for instance, and lack of
+##  implementations when View exists, but ViewString doesn't (probably for the
+##  reason above)
+##
+##  output streams will be free to ignore/strip formatting helpers
+##
+##  The naming is still topic of discussion. I renamed Print to Code, because
+##  this makes the requirement to output Code clearer.
+##
+
+############################################################################
+##
+#O  ViewObjStream(<stream>, <obj>)
+##
+InstallMethod(ViewObjStream, "for an output stream, and an object",
+              [IsOutputStream, IsObject],
+function(stream, obj)
+    PrintTo(stream, ViewString(obj));
+end);
+
+# A possible implementation for ViewString
+# (once ViewObjStream is the default)
+NewViewString := function(obj)
+   local res, stream;
+   res := "";
+   stream := OutputTextString(res, true);
+   ViewObjStream(stream, obj);
+   return res;
+end;
```

Comment:
I know this is just a placeholder, but: maybe call this `AlternateViewString`, to avoid confusion ("hu, a function which creates a new ViewString?")

Also, we could add an `AlternateString` function, which uses a special text string stream that strips out `\<` (alternatively, we could just strip them out after creating the print string, but that may be slower for some kinds of output -- say, if the final string is huge, but each piece we add to it incrementally is relatively small -- then we just need to remove \< from inside small pieces, instead of removing them from inside a single huge string.

(Or, with the alternative proposal to not use `\<` at all, we could of course save even more.

Reply:
What I am missing here is a clear migration plan, though. It might go like this: We add this `NewViewString` as a new method for `ViewString` for now, with very low rank, replacing the current default method (which returns `DEFAULTVIEWSTRING`). That also means that we'd have to provide a default method for `ViewObjStream`, which prints `DEFAULTVIEWSTRING` to the output stream.

The same would be needed for `PrintString` and `DisplayString`, of course, and ideally also `PrintObj` and `Display`, but we'd have to be careful about not breaking the existing (somewhat complicated) delegation rules.
(And once this setup is complete, the documentation needs to be updated).

Of course we could also do something else; but we need a plan for an *incremental* conversion, before we start, else this will end up in the same incomplete, semi-working and overall mostly useless state as its predecessor project, i.e. the introduction of `ViewString` and friends.

Once we have a plan we think is fine, we then should try to migrate at least the library to the new system ASAP, and also aggressively push package authors to do the same (at least once a GAP release with the new system has been released).